server/src/test/java/org/elasticsearch/repositories/blobstore/ShardSnapshotTaskRunnerTests.java

```diff
@@ -52,12 +53,16 @@ public void setUp() throws Exception {
         super.setUp();
         threadPool = new TestThreadPool("test");
         executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
+        dummyStores = new ArrayList<>();
     }
 
     @Override
     public void tearDown() throws Exception {
         super.tearDown();
         TestThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        // Don't trigger leak detection
+        dummyStores.forEach(Store::close);
+        dummyStores.clear();
```

Comment:
I think this probably does not work since the `dummyContext` helper method is also used by a different test class `BlobStoreRepositoryTests`. One possible solution is to let the caller manages it. The helper method returns a `SnapshotShardContext` which has `store()` method to access the store and close it?